File: Java_Dsc/Stack.java
/* Java implemtation of Stack Data Structure */
class Stack
{
    static final int MAX = 1000;
    int top;
    int a[] = new int[MAX];

    boolean isEmpty(){
        return (top < 0);
    }

    boolean push(int x){
        if(top >= MAX){
            System.out.println("Stack Underflow");
            return false;
        } else {
            a[++top] = x;
            return true;
        }
    }

    int pop(){
        if(top<0){
             System.out.println("Stack Underflow");
             return 0;
        } else {
            int x = a[top--];
            return x;
        }
    }


    public static void main(String args[])
    {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.push(30);
        System.out.println(s.pop() + " Popped from stack");
    }

}

/*Output:
 * 30 Popped from stack
 */

File: Python_Dsc/Stack.py
# Python implementation of Stack Datastructure

#Used to return -infinite when stack is empty
from sys import maxsize

def createStack():
    stack = []
    return stack

def isEmpty(stack):
    return len(stack) == 0

def push(stack, item):
    stack.append(item)
    print("pushed to stack"+item)

def pop(stack):
    if(isEmpty(stack)):
        return str(-maxsize -1) #return minus infinite
    
    return stack.pop()

# Driver program to test above functions    
stack = createStack()
push(stack, str(10))
push(stack, str(20))
push(stack, str(30))
print(pop(stack) + " popped from stack")

''' Output:
    pushed to stack10
    pushed to stack20
    pushed to stack30
    30 popped from stack
    '''
